client.js

```diff
@@ -0,0 +1,145 @@
+var scene = new THREE.Scene();
+var camera = new THREE.PerspectiveCamera( 75, window.innerWidth / window.innerHeight, 0.1, 1000 );
+
+var renderer = new THREE.WebGLRenderer();
+renderer.setSize( window.innerWidth, window.innerHeight );
+document.body.appendChild( renderer.domElement );
+
+// arrow rendering parameters
+var maxArrowLength = 0.375;
+var hex = 0xffff00;
+var maxHeadLength = 0.5 * maxArrowLength;
+var maxHeadWidth = 0.5 * maxArrowLength;
+
+camera.position.z = 10;
+
+class Plotter {
+    constructor() {
+        this.vectorFieldSamples = [];
+
+        this.chargeGeometry = new THREE.CircleGeometry( 0.1, 32 );
+        this.chargeMaterial = new THREE.MeshBasicMaterial( { color: 0x0000FF} );
+        this.pointCharge = new THREE.Mesh( this.chargeGeometry, this.chargeMaterial );
+        scene.add( this.pointCharge );
+    }
+
+    updatePlot(newVectorFieldSamples, newChargePosition) {
+        for (var arrow of this.vectorFieldSamples) {
+            scene.remove(arrow);
+        };
+
+        this.vectorFieldSamples = [];
+
+        for (var newSample of newVectorFieldSamples) {
+            var origin = new THREE.Vector3(newSample[0][0], newSample[0][1], newSample[0][2]);
+            var dir = new THREE.Vector3(newSample[1][0], newSample[1][1], newSample[1][2]);
+            var vectorLength = dir.length();
+            // convert the vectorLength to a number between 0 and 1. That way plotted vector length
+            // isn't too large.
+            var strengthFactor = 1 - 1/(1.0 + 100*Math.log(1 + vectorLength))
+            dir.normalize();
+
+            var newArrow = new THREE.ArrowHelper(dir, origin, maxArrowLength * strengthFactor, hex, 
+                maxHeadLength * strengthFactor, maxHeadWidth * strengthFactor);
+            newArrow.cone.material.opacity = strengthFactor;
+            newArrow.line.material.opacity = strengthFactor;
+            newArrow.cone.material.transparent = true;
+            newArrow.line.material.transparent = true;
```

Comment:
This block does 1 thing. Could extract into a separate function: `buildArrayFromNewSample`